src/kdumpfile/s390x.c

```diff
@@ -230,7 +230,119 @@ s390x_init(kdump_ctx_t *ctx)
 	return KDUMP_OK;
 }
 
+struct elf_siginfo
+{
+	int32_t si_signo;	/* signal number */
+	int32_t si_code;	/* extra code */
+	int32_t si_errno;	/* errno */
+} __attribute__((packed));
+
+struct psw {
+	unsigned long mask;
+	unsigned long addr;
+} __attribute__((packed));
+
+struct elf_s390_regs {
+	struct psw psw;
+	unsigned long gprs[16];
```

Comment:
Please do not use unsigned long here. The structure layout is part of the target ABI. Since libkdumpfile is designed to be cross-platform, target data structures must not depend on host types.

Reply:
Of course. I change that to uint64_t and uint32_t.